src/box/index.h

```diff
@@ -459,23 +459,23 @@ key_validate(const struct index_def *index_def, enum iterator_type type,
 	     const char *key, uint32_t part_count);
 
 /**
- * Check that the supplied key is valid for a search in a unique
- * index (i.e. the key must be fully specified).
+ * Check that the supplied primary index key is valid for a search in
+ * a unique index (i.e. the key must be fully specified).
  * @retval 0  The key is valid.
  * @retval -1 The key is invalid.
  */
 int
-exact_key_validate(struct key_def *key_def, const char *key,
+exact_key_validate(struct index_def *index_def, const char *key,
 		   uint32_t part_count);
 
 /**
- * Check that the supplied key is valid for representing iterator
+ * Check that the supplied user-defined key is valid for representing iterator
```

Comment:
Can't catch why you changed the comment here. I'm not sure that the function is used only for user-defined key, perhaps it is or will be used for developer-defined keys.

Reply:
As we've discussed privately, I changed `exact_key_validate_nullable()` to check `index_def->cmp_def` that is according to description a user defined key.

```C
         /**
           * User-defined key definition, merged with the primary
           * key parts. Used by non-unique keys to uniquely identify
           * iterator position.
           */
          struct key_def *cmp_def;
```

Reply:
Ok, we've discussed it once more and I reworded the comments.

```diff
 /**
- * Check that the supplied key is valid for a search in a unique
- * index (i.e. the key must be fully specified).
+ * Check that the index_def->key_def is valid for a search in
+ * a unique index (i.e. the key must be fully specified).
  * @retval 0  The key is valid.
  * @retval -1 The key is invalid.
  */
 int
-exact_key_validate(struct key_def *key_def, const char *key,
+exact_key_validate(struct index_def *index_def, const char *key,
                   uint32_t part_count);
 
 /**
- * Check that the supplied key is valid for representing iterator
+ * Check that the index_def->cmp_def is valid for representing iterator
  * position (i.e. the key must be fully specified, but nulls are allowed).
  * @retval 0  The key is valid.
  * @retval -1 The key is invalid.
  */
 int
-exact_key_validate_nullable(struct key_def *key_def, const char *key,
+exact_key_validate_nullable(struct index_def *index_def, const char *key,
```